src/Storage/Storage/Common/AzureDataLakeGen2Item.cs

```diff
@@ -0,0 +1,144 @@
+
+        /// <summary>
+        /// CloudBlobDirectory object
+        /// </summary>
+        public CloudBlobDirectory CloudBlobDirectory { get; private set; }
+
+
+        /// <summary>
+        /// The Path of the item
+        /// </summary>
+        [Ps1Xml(Label = "Path", Target = ViewControl.Table, Position = 0, TableColumnWidth = 20)]
+        public string Path { get; set; }
+
+        /// <summary>
+        /// CloudBlobDirectory object
+        /// </summary>
+        [Ps1Xml(Label = "IsDirectory", Target = ViewControl.Table, Position = 1, TableColumnWidth = 12)]
+        public bool IsDirectory { get; private set; }
+
+        /// <summary>
+        /// Datalake Gen2 Item path Permissions
+        /// </summary>
+        [Ps1Xml(Label = "Permissions", Target = ViewControl.Table, ScriptBlock = "$_.Permissions.ToSymbolicString()", Position = 5, TableColumnWidth = 12)]
+        public PathPermissions Permissions { get; set; }
+
+        /// <summary>
+        /// Datalake Gen2 Item ACL
+        /// </summary>
+        public PSPathAccessControlEntry[] ACL { get; set; }
+
+        /// <summary>
+        /// Azure storage blob type
+        /// </summary>
+        public BlobType BlobType { get; private set; }
+
+        /// <summary>
+        /// Blob length
+        /// </summary>
+        [Ps1Xml(Label = "Length", Target = ViewControl.Table, ScriptBlock = "if ($_.IsDirectory -eq $false) {$_.Length}", Position = 2, TableColumnWidth = 15)]
+        public long Length { get; private set; }
+
+        /// <summary>
+        /// Blob content type
+        /// </summary>
+        [Ps1Xml(Label = "ContentType", Target = ViewControl.Table, Position = 3, TableColumnWidth = 30)]
+        public string ContentType { get; private set; }
+
+        /// <summary>
+        /// Blob last modified time
+        /// </summary>
+        [Ps1Xml(Label = "LastModified", Target = ViewControl.Table, ScriptBlock = "$_.LastModified.UtcDateTime.ToString(\"u\")", Position = 4, TableColumnWidth = 20)]
+        public DateTimeOffset? LastModified { get; private set; }
+
+        /// <summary>
+        /// The owner of the item
+        /// </summary>
+        [Ps1Xml(Label = "Owner", Target = ViewControl.Table, Position = 6, TableColumnWidth = 10)]
+        public string Owner { get; set; }
+
+        /// <summary>
+        /// The Group of the item
+        /// </summary>
+        [Ps1Xml(Label = "Group", Target = ViewControl.Table, Position = 7, TableColumnWidth = 10)]
+        public string Group { get; set; }
+
+        /// <summary>
+        /// Blob continuation token
+        /// </summary>
+        public BlobContinuationToken ContinuationToken { get; set; }
+
+        /// <summary>
+        /// Azure DataLakeGen2 Item constructor
+        /// </summary>
+        /// <param name="blob">CloudBlockBlob blob object</param>
+        public AzureDataLakeGen2Item(CloudBlockBlob blob)
+        {
+            Name = blob.Name;
```

Comment:
Is `blob.Name` the filename or the path or the file?

Reply:
Actually, the blob name is the full blob path under the container , which is just aligned with path.